commerce_coordinator/apps/orders/clients.py

```diff
@@ -0,0 +1,82 @@
+
+
+class BaseEdxOAuthClient:
+    """
+    API client for calls to the other edX services.
+    """
+
+    def __init__(self):
+        self.client = OAuthAPIClient(
+            settings.SOCIAL_AUTH_EDX_OAUTH2_URL_ROOT.strip('/'),
+            self.oauth2_client_id,
+            self.oauth2_client_secret,
+            timeout=(
+                settings.REQUEST_CONNECT_TIMEOUT_SECONDS,
+                settings.REQUEST_READ_TIMEOUT_SECONDS
+            )
+        )
+
+    @property
+    def oauth2_client_id(self):
+        return settings.BACKEND_SERVICE_EDX_OAUTH2_KEY
+
+    @property
+    def oauth2_client_secret(self):
+        return settings.BACKEND_SERVICE_EDX_OAUTH2_SECRET
+
+
+class EcommerceApiClient(BaseEdxOAuthClient):
+    """
+    API client for calls to the edX Ecommerce service.
+    """
+    api_base_url = str(settings.ECOMMERCE_URL) + '/api/v2/'
+
+    def get_orders(self, query_params):
+        """
+        Call ecommerce API overview endpoint for data about an order.
+
+        Arguments:
+            username: restrict to orders by this username
+        Returns:
+            dict: Dictionary represention of JSON returned from API
+
+        example response:
+        {
+            {"count": 1, "next": null, "previous": null, "results": [{"billing_address": {"first_name":
+            "Diane", "last_name": "Test", "line1": "50 turner st", "line2": "", "postcode": "02135",
+            "state": "MA", "country": "US", "city": "Brighton"}, "currency": "USD", "date_placed":
+            "2021-12-20T15:09:44Z", "discount": "0", "lines": [{"title": "Seat in edX Demonstration
+            Course with verified certificate (and ID verification)", "quantity": 1, "description":
+            "Seat in edX Demonstration Course with verified certificate (and ID verification)", "status":
+            "Complete", "line_price_excl_tax": "149.00", "unit_price_excl_tax": "149.00", "product":
+            {"id": 3, "url": "http://localhost:18130/api/v2/products/3/", "structure": "child",
+            "product_class": "Seat", "title": "Seat in edX Demonstration Course with verified certificate
+            (and ID verification)", "price": "149.00", "expires": "2022-11-08T22:54:30.777313Z",
+            "attribute_values": [{"name": "certificate_type", "code": "certificate_type", "value":
+            "verified"}, {"name": "course_key", "code": "course_key", "value":
+            "course-v1:edX+DemoX+Demo_Course"}, {"name": "id_verification_required", "code":
+            "id_verification_required", "value": true}], "is_available_to_buy": true, "stockrecords":
+            [{"id": 3, "product": 3, "partner": 1, "partner_sku": "8CF08E5", "price_currency": "USD",
+            "price_excl_tax": "149.00"}]}}], "number": "EDX-100004", "payment_processor":
+            "cybersource-rest", "status": "Complete", "total_excl_tax": "149.00", "user":
+            {"email": "edx@example.com", "username": "edx"}, "vouchers": []} ]}
+        }
```

Comment:
I'd format this to be more readable or refer the reader to the fixture in `tests.py` (the former is convenient for the reader, but more likely to get stale; as-written, the outermost curly-braces should probably be square-brackets)

Reply:
Good call- this shouldn't be repeated! I updated the comment to refer to the sample_response in tests.py if the reader wants to see an example, good call